src/js/components/List/List.js

```diff
@@ -319,7 +320,15 @@ const List = React.forwardRef(
                   content = item;
                 }
 
-                const key = getKey(item, index, itemId);
+                if (itemKey) {
+                  if (typeof itemKey === 'function') {
+                    itemId = itemKey(item);
+                  } else {
+                    itemId = normalize(item, index, itemKey);
+                  }
+                }
+
+                const key = itemId || getKey(item, index, itemId);
```

Comment:
```suggestion
                const key = itemKey ? itemId : getKey(item, index, itemId);
```
I think we should change this so that we only set key to itemId when we have an `itemKey` defined. Otherwise we get an error in the case where we don't have an `itemKey` but we have a `primaryKey` that is a function and it tries to set the key to the `primaryKey` function.